### ISSUE TITLE
[Component] [autocomplete] autocomplete closes automatically when clicking on a custom element 

### ISSUE THREAD

**Issue by arthabus** [element-plus/element-plus#17171]
Bug Type: **`Component`**

## Environment

- Vue Version: `3.3.4`
- Element Plus Version: `2.3.14`
- Browser / OS: `Version 116.0.5845.187 (Official Build) (x86_64)`
- Build Tool: `Webpack`

### Related Component

- `el-autocomplete`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6IlxuPHRlbXBsYXRlPlxuICA8ZWwtYXV0b2NvbXBsZXRlIFxuICAgIHBsYWNlaG9sZGVyPVwiY2xpY2sgaGVyZVwiXG4gICAgOmZldGNoLXN1Z2dlc3Rpb25zPVwicXVlcnlTZWFyY2hBc3luY1wiXG4gICAgdi1tb2RlbD1cInNlbGVjdGVkXCI+XG4gIFx0XHQ8dGVtcGxhdGUgI2RlZmF1bHQ9XCJ7IGl0ZW0gfVwiPlxuICAgICAgICA8ZGl2Pnt7aXRlbX19PGVsLWJ1dHRvbiBAY2xpY2suc3RvcD1cIigpID0+IG9uQ2xpY2tCdG4oaXRlbSlcIj5DbGljayBNZTwvZWwtYnV0dG9uPjwvZGl2PlxuXHRcdDwvdGVtcGxhdGU+XG4gIDwvZWwtYXV0b2NvbXBsZXRlPlxuXG48L3RlbXBsYXRlPlxuXG48c2NyaXB0PlxuXG5leHBvcnQgZGVmYXVsdCB7XG5cblxuICBkYXRhKCl7XG4gICAgcmV0dXJuIHtcbiAgICAgIHNlbGVjdGVkOiBudWxsLFxuICAgICAgb3B0aW9uczogW1wib3B0aW9uMVwiLCBcIm9wdGlvbjJcIl1cbiAgICB9XG4gIH0sXG5cbiAgbWV0aG9kczoge1xuICAgIHF1ZXJ5U2VhcmNoQXN5bmMocXVlcnlTdHJpbmcsIGNhbGxiYWNrKXtcbiAgICAgIGNhbGxiYWNrKHRoaXMub3B0aW9ucylcbiAgICB9LFxuICAgIG9uQ2xpY2tCdG4oaXRlbSl7XG4gICAgY29uc29sZS5sb2coXCJidXR0b24gY2xpY2tlZCBmb3IgaXRlbTogXCIsIGl0ZW0pXG4gICAgfVxuICB9XG59XG48L3NjcmlwdD5cblxuXG5cblxuXG5cblxuXG5cblxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJ0c2NvbmZpZy5qc29uIjoie1xuICBcImNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcImpzeFwiOiBcInByZXNlcnZlXCIsXG4gICAgXCJtb2R1bGVcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVJlc29sdXRpb25cIjogXCJCdW5kbGVyXCIsXG4gICAgXCJ0eXBlc1wiOiBbXCJlbGVtZW50LXBsdXMvZ2xvYmFsLmQudHNcIl0sXG4gICAgXCJhbGxvd0ltcG9ydGluZ1RzRXh0ZW5zaW9uc1wiOiB0cnVlLFxuICAgIFwiYWxsb3dKc1wiOiB0cnVlLFxuICAgIFwiY2hlY2tKc1wiOiB0cnVlXG4gIH0sXG4gIFwidnVlQ29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiAzLjNcbiAgfVxufVxuIiwiX28iOnt9fQ==)

### Steps to reproduce

1. Click the input field
2. Click the "click me" button from the popup

### What is Expected?

Popup stays open

### What is actually happening?

Popup closes automatically

### Additional comments

This worked well in the vue2 element version.

After digging through the source code of the vue3 element I can see that the root cause of this is onClickOutside which is triggered even when clicked on a custom element inside of the popup.

Providing an option to disable onClickOutside would solve it

**Comment by arthabus** [comment]
Submitted a PR https://github.com/element-plus/element-plus/pull/17179 with the fix. Please someone review

**Comment by mbokil** [comment]
The history on the autocomplete component is funny. 2 years ago I saw a commit to auto close the popup as a bug fix. Even though it creates problems for me I think auto closing of the popup is what most users expect. I suggest rejecting this PR and instead implement a new boolean value :auto-close="false" that way it won't break user's current behavior but will give devs an option for more complex custom templates where they need to control the popup closing.

**Comment by arthabus** [comment]
@mbokil I agree that closing a popup when option is selected is the default expected behavior. 

The issue described here is a slightly different one though: each option can be customized using slot and when there is a custom clickable element, like a button, inside of option which is clicked, this is when I expect to be in control of closing the popup. 

This doesn't overrides the default behavior, but extends it - if user clicks an option, the popup closes. Only when they click a custom button inside of such option, the decision should be for developer to close it or not.  This is also how it worked in Vue2 version.

**Comment by tolking** [comment]
This issue has been fixed. The dropdown panel will not close as long as the focus remains on it.
